jetty-websocket/websocket-jetty-api/src/main/java/org/eclipse/jetty/websocket/api/WebSocketAdapter.java

```diff
@@ -21,10 +21,11 @@
 public class WebSocketAdapter implements WebSocketListener
 {
     private volatile Session session;
+    private RemoteEndpoint remote;
 
     public RemoteEndpoint getRemote()
     {
-        return session.getRemote();
+        return remote;
```

Comment:
IMO `return session == null ? null : session.getRemote();` might be better. This is minor, but could save the `remote` field. And I think the readability is not affected too much either way.

Reply:
It would be more surprising for `getRemote()` to suddenly start returning `null` after it was non-null.

Let the usage of `Remote` trigger the exceptions indicating that you no longer have a valid connection/session/whatnot.